src/components/AutoSaveIndicator.tsx

```diff
@@ -0,0 +1,34 @@
+import { AutoSaveStatus, useAutoSaveStatus } from "@homebound/form-state";
+import { assertNever } from "@homebound/form-state/dist/utils";
+import { useEffect } from "react";
+import { Tooltip } from ".";
+import { Palette } from "..";
+import { Icon } from "./Icon";
+
+interface AutoSaveIndicatorProps {
+  hideOnIdle?: boolean;
+}
+
+export function AutoSaveIndicator({ hideOnIdle }: AutoSaveIndicatorProps) {
+  const { status, resetStatus, errors } = useAutoSaveStatus();
+
+  /** On Dismount, reset back to Idle so new pages don't imply something "saved" */
+  useEffect(() => () => resetStatus(), []);
```

Comment:
thought: Seems like this shouldn't be the indicators responsibility, but the responsibility of anyone who's called `triggerAutoSave`, that has an outstanding save when they're unmounted, to resolve it / not leaving it dangling?

I.e. could `useAutoSaveStatus` just do this? Recognize when its component has called `triggerAutoSave` --> no resolve --> its component is dismounted, it auto-resolves it? 

(Like I could see a component hierarchy where the AutoSaveIndicator dismounts, but some sibling component still has valid/outstanding saves, live in the shared AutoSaveContext. Granted, this setup seems a little weird, insofar as I think the assumption you have here of "the AutoSaveIndicator lives above all other auto save components in this AutoSaveContext, so the indicator unmounting --> defacto the others have unmounted" is intuitive/probably how it'd be setup, but not necessarily _has_ to be / will be guaranteed afaiu.)

Reply:
I'd argue that there are plenty of components that _could_ call `useAutoSave()` just to observe behavior and they wouldn't want "side effects" from calling that hook. Form's only interest is Triggering/Resolving. It does not care if things are reset or what the active state of the button is. Just trigger and resolve.

AutoSaveIndicator, however, cares a lot about Status. It displays based on it. If * it * dismounts, it's almost definitely for a determinable reason (closed modal, page navigation, etc) and given that it's the only component that actually cares about what the Status is, it also adopts the responsibility to reset it at will.

Reply:
> wouldn't want "side effects" from calling that hook

Of course, my suggestion is only for components like forms who've explicitly `triggerAutoSave`-d.

The contract would be "if you trigger auto save, it's on you to either resolve it explicitly or implicitly on unmount".

> given that it's the only component that actually cares about what the Status is,
> it also adopts the responsibility to reset it at will.

This articulation just seems wrong. This component being the only one that cares about _reading_ status (to display) IMO doesn't follow that it gets to decide to _write_ status as well.

Reply:
Can we find a middleground here, please: https://github.com/homebound-team/beam/pull/550/commits/4a21ef1f8b2d53645575227d323bd1a5d704b490

Reply:
Has this behavior regarding the status needing to be reset vs persist across pages about been discussed with our UX/Design team? I think I've mentioned this before, but maybe just to you @TylerR909, so I'll say it again here... If that status persists, I don't think thats a terrible thing. With SPAs we kind of tend to want less 'reset everything' as the user navigates, and make it feel more like a cohesive experience. Though, the placement of the indicator could make it confusing if you were on Homeowner Contracts and then switched to Commitments as it'd show that you "saved" changes in Homeowner Contracts, then "saved" may still show next to the Commitments title - and that doesn't seem right. But if you were in Commitments and you switch between the Line Items tab to the Overview tab and the "Saved" indication persisted there, I think that's fine. It'll still show that you "saved" something related to Commitments.
Thinking through those use cases, I'd tend to want the caller to dictate when to reset the status. In the instance of switching between the tabs, then maybe the page that builds the tabs would have a `useEffect` that only resets the status on unmount. And the content within the tabs could trigger the autosave. 
I guess to distill it down/TLDR, I think if the content around the AutoSaveIndicator changes - like the title of the page/modal/drawer, then we'd want to reset it. But if the title remains and we change tabs within the page, I think it would be a-okay to persist it.
Maybe the `useEffect` for reset on unmount something we just do in the PageHeader, ModalHeader, Drawer header components? 
Anyways, these sort of behaviors in the application should be a discussion with UX/Design if it hasn't been already. Present the different use cases, pros & cons, and come to consensus.

Reply:
Jet had a 10-minute with Jeffrey and we stepped through some pages & drawers thinking when it should vs shouldn't reset.

Long story short: Any time this title changes, it should reset:
<img width="422" alt="image" src="https://user-images.githubusercontent.com/20718430/172492589-98d8ce29-8b17-4ee8-a96d-0665dd1285ad.png">

1. Title doesn't change in "Tabbed" pages like the Commitments Editor
2. Title does change, and should reset, any time you click up and down the Project Navbar
3. A drawer opening (and a new AutoSaveIndicator mounting) should NOT indicate anything in the drawer has saved yet (i.e. should reset on mount)
3b. He doesn't care if a drawer opening resets an AutoSave in the background
3c. When you close the drawer it may also reset

To me, this effectively boils down to any time it Mounts or Dismounts, reset.

It's confusing to users that they'd go to a new page and BluePrint would be telling them something on that page or in a newly-opened Drawer has been "Saved" before they've even touched anything.

Reply:
Drawers would have their own AutoSaveContext, so that makes sense (an action auto-saving in the drawer shouldn't make my greyed-out page header AutoSaveIndicator start spinning).

Actually probably same with pages (seems like Jeff is asking for them conceptually to have their own individual AutoSaveContexts), but we could probably have a global one and call it good enough, with using this heuristic of "however high up the component tree you put the AutoSaveIndicator, you're probably meaning to have an AutoSaveContext basically mapped 1-1 to that context, so AutoSaveIndicator unmount --> probably means 'sure a new/reset context'".

I.e., thinking through the alternative, if we really did have a single global AutoSaveIndicator up in the top-left, our pages would defacto share an AutoSaveContext, b/c moving from page 1 (with a save in flight) to page 2, I assume, shouldn't flash-/immediate-reset the global AutoSaveIndicator. We'd want it to finish it's current spinning w/o jarring away.

So anyway, with that clarification/articulation in my that, that personally "when do I reset" should not really be a question/operation, and instead "where do you want resets?" is more about "where do you want to draw context lines in your app", and then you would explicitly have a 1-1 mapping between AutoSaveContext <-> AutoSaveIndicator, and it's kind of a non-issue.

_But_, if we want to roll with a single global AutoSaveContext, and use the heuristic of AutoSaveIndicator unmounts as a hint to "we're effectively moving to a new context", I'm good with that.

Reply:
Great! Thanks for taking the time to work through the behavior with Jeff.